File: HealthAxis_Web/Models/Appointment.cs
using HealthAxis.Shared.Dtos;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HealthAxis.Api.Models
{
    public class Appointment
    {
        [Key]
        public int AppointmentId { get; set; }

        [ForeignKey("Patient")]
        public int PatientId { get; set; }
        public Patient Patient { get; set; }

        [ForeignKey("Doctor")]
        public int DoctorId { get; set; }
        public Doctor Doctor { get; set; }

        [Required]
        public DateTime ScheduledDate { get; set; }

        [Required]
        [StringLength(20)]
        public string TimeSlot { get; set; }

        [Required]
        public string Status { get; set; }

        [StringLength(500)]
        public string CancellationReason { get; set; }
    }
}

File: HealthCareAppPatientDoctor.Angular/src/app/core/models/appointment.ts
export interface Appointment {

  appointmentId: number;

  patientId: number;

  patientName?: string;

  doctorId: number;

  doctorName?: string;

  scheduledDate: string;

  timeSlot: string;

  status: number;

  cancellationReason?: string;

}